schematicbrushreborn-core/src/main/java/de/eldoria/schematicbrush/config/sections/presets/DatabasePresetRegistry.java

```diff
@@ -0,0 +1,284 @@
+package de.eldoria.schematicbrush.config.sections.presets;
+
+import com.google.gson.Gson;
+import de.eldoria.eldoutilities.serialization.SerializationUtil;
+import de.eldoria.eldoutilities.simplecommands.TabCompleteUtil;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+public class DatabasePresetRegistry implements PresetRegistry {
+
+    private static final Gson GSON = new Gson();
+    private Map<UUID, PresetContainer> playerPresets = new HashMap<>();
+    private PresetContainer globalPresets = new PresetContainerImpl();
+    private final DataSource dataSource;
+    private final Plugin plugin;
+
+    public DatabasePresetRegistry(DataSource dataSource, Plugin plugin) {
+        this.dataSource = dataSource;
+        this.plugin = plugin;
+        this.reload();
+    }
+
+    private void reload() {
+        this.playerPresets.clear();
+        this.globalPresets.getPresets().clear();
+        CompletableFuture.runAsync(() -> {
+            try(Connection connection = this.dataSource.getConnection()) {
```

Comment:
In general setup logic should be located in a setup file and not embedded in the code